### PR TITLE
[FIX] website_links: fix multiple bugs + add tests

### PR DESCRIPTION
**Pull request by rdeodoo**
Commit 6d67ce5b7c refactored the JS from multiple website modules including
`website_links`.
A few error needed to be fixed:
  - recent links would not appear anymore
  - graphs would not be displayed anymore
  - graph resize on tab change would not be fired

This commit fixes it and also add some tests to be sure this module is now
strongly tested.

### CODE REVIEW

**Comment by rdeodoo** [comment]
@qsm-odoo You will see 2 TODO left in the code for which I need your input.

**Comment by rdeodoo** [comment]
@JKE-be This comes from CLO as she is using the website_links module on odoo.com, and since the migration she had some issues.

She also had some questions (that's why I am pinging you here) for which you probably already have the answer, or that probably already have been debated:
1. Recent links in frontend /r are limited to 20, so no way to retrieve some links that you want to (eg to edit the URL since you can't in the backend). It is probably critical for the ones using the module. Feedback? Backlog task to create?
2. Link tracker code [are case sensitive](https://github.com/odoo/odoo/blame/7cdf8625bcc32650f6724029091722abc9e13d06/addons/link_tracker/models/link_tracker.py#L198), is this to avoid collision (given the code that [generate the code](https://github.com/odoo/odoo/blame/7cdf8625bcc32650f6724029091722abc9e13d06/addons/link_tracker/models/link_tracker.py#L227) I think it could be case insensitive if we replace search `=` by `=ilike` as the generation of the code is already ensure to not have collision)?
In her example, the link is something like "POSmodule" to print on a flyer. That way POS in capital make sense for people to understand the meaning but they will probably not write POS in capital when accessing the link.
Anyway it is weird to have case sensitive URL IMO.
3. Why is it not possible in the backend to change the code while in the frontend it is? Wanted behavior?

File: addons/website_links/static/src/js/test_ui_flow.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+odoo.define('website_links.website_links_tour', function (require) {
+'use strict';
+
+var tour = require("web_tour.tour");
+var base = require("web_editor.base");
```
Comment (qsm-odoo) [inline addons/website_links/static/src/js/test_ui_flow.js]: Does not exist anymore in master... mmh.

File: addons/website_links/static/src/js/test_ui_flow.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            content: "fill the form and submit it",
+            trigger: '#o_website_links_link_tracker_form input#url',
+            run: function () {
+                var url = window.location.host + '/aboutus';
+                $('#o_website_links_link_tracker_form input#url').val(url);
+                $('.o_website_links_utm_forms input#campaign-select').val(1).change();
+                $('.o_website_links_utm_forms input#channel-select').val(1).change();
+                $('.o_website_links_utm_forms input#source-select').val(1).change();
+                $('#btn_shorten_url').click();
+            }
```
Comment (qsm-odoo) [inline addons/website_links/static/src/js/test_ui_flow.js]: Missing trailing comma :D

File: addons/website_links/static/src/js/website_links_charts.js [hunk 3]
```diff
@@ -305,7 +305,12 @@ sAnimations.registry.websiteLinksCharts = sAnimations.Class.extend({
     _onGraphTabClick: function (ev) {
         ev.preventDefault();
         $('.graph-tabs li a').tab('show');
-        _.chain(this.charts).pluck('chart').invoke('update'); // Force NVD3 to redraw the chart
```
Comment (qsm-odoo) [inline addons/website_links/static/src/js/website_links_charts.js]: Strange... this line did not work in 12.0 so ?

Comment (rdeodoo) [reply addons/website_links/static/src/js/website_links_charts.js]: It was working in 12.0
Somehow it is not working anymore on saas-12.1:

screenshot from 2019-03-01 11-57-20

The weird part is that on the `start` method, there is a
```javascript
nv.utils.windowResize(function () {
    _.chain(self.charts).pluck('chart').invoke('update');
});
```
That works well, if you resize your windows the charts are correctly redrawn..
But on this `_onGraphTabClick` the same line is not working anymore, even in a setTimeout.

Anyway I googled a bit and found these solutions:
```javascript
window.dispatchEvent(new Event('resize'));
```
or
```javascript
$(window).trigger('resize');
```
None is working, except the first if wrapped into a `setTimeout(0)`..

File: addons/website_links/static/src/js/test_ui_flow.js (hunk 2 above)
Comment (rdeodoo) [reply addons/website_links/static/src/js/test_ui_flow.js]: Done

**Comment by rdeodoo** [comment]
As discussed, I kept the setTimeout(0)  and applyed the small changes.
Code is clean now.

File: addons/website_links/views/website_links_template.xml [hunk 4]
```diff
@@ -6,6 +6,7 @@
                 <script type="text/javascript" src="/website_links/static/src/js/website_links.js"/>
                 <script type="text/javascript" src="/website_links/static/src/js/website_links_code_editor.js"/>
                 <script type="text/javascript" src="/website_links/static/src/js/website_links_charts.js"/>
+                <script type="text/javascript" src="/website_links/static/src/js/test_ui_flow.js"></script>
```
Comment (qsm-odoo) [inline addons/website_links/views/website_links_template.xml]: And I would do the same as the above here and autoclose the script tag :)

Comment (rdeodoo) [reply addons/website_links/views/website_links_template.xml]: Done

**Comment by rdeodoo** [comment]
@robodoo r+
I skipped the test for now and will check ASAP that race condition issue appearing only on Docker. Once done I will re-enable the test.

**Pull request closed without merging**